SIGSEGV 0.33.4

`helmfile -e stage sync`

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x766e6b]

goroutine 1 [running]:
github.com/roboll/helmfile/vendor/github.com/urfave/cli.HandleAction.func1(0xc420154f48)
	/go/src/github.com/roboll/helmfile/vendor/github.com/urfave/cli/app.go:474 +0x279
panic(0x7d10a0, 0xa79ae0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
main.(*app).FindAndIterateOverDesiredStates(0xc420154a80, 0x0, 0x0, 0xc420154af0, 0x0, 0x0, 0xaa6948, 0x0, 0x0, 0x7fffd6ed9098, ...)
	/go/src/github.com/roboll/helmfile/main.go:630 +0x4db
main.findAndIterateOverDesiredStatesUsingFlags(0xc4200d4c80, 0xc420154af0, 0xc420154b10, 0x16)
	/go/src/github.com/roboll/helmfile/main.go:568 +0x2f3
main.main.func6(0xc4200d4c80, 0x0, 0x0)
	/go/src/github.com/roboll/helmfile/main.go:273 +0x46
reflect.Value.call(0x7b7380, 0x859aa0, 0x13, 0x83a5cf, 0x4, 0xc420154ee8, 0x1, 0x1, 0xc42000e140, 0x82de60, ...)
	/usr/local/go/src/reflect/value.go:447 +0x969
reflect.Value.Call(0x7b7380, 0x859aa0, 0x13, 0xc420154ee8, 0x1, 0x1, 0x4, 0xc420095350, 0x5)
	/usr/local/go/src/reflect/value.go:308 +0xa4
github.com/roboll/helmfile/vendor/github.com/urfave/cli.HandleAction(0x7b7380, 0x859aa0, 0xc4200d4c80, 0x0, 0x0)
	/go/src/github.com/roboll/helmfile/vendor/github.com/urfave/cli/app.go:483 +0x173
github.com/roboll/helmfile/vendor/github.com/urfave/cli.Command.Run(0x83ab63, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x85390a, 0x43, 0x0, ...)
	/go/src/github.com/roboll/helmfile/vendor/github.com/urfave/cli/command.go:186 +0x999
github.com/roboll/helmfile/vendor/github.com/urfave/cli.(*App).Run(0xc420001200, 0xc42000e0c0, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/roboll/helmfile/vendor/github.com/urfave/cli/app.go:237 +0x546
main.main()
	/go/src/github.com/roboll/helmfile/main.go:474 +0x1f79
```

--- a/app_test.go
+++ b/app_test.go
@@ -106,3 +106,95 @@ releases:
 		}
 	}
 }
+
+// See https://github.com/roboll/helmfile/issues/320
+func TestFindAndIterateOverDesiredStates_UndefinedEnv(t *testing.T) {
+	absPaths := map[string]string{
+		".": "/path/to",
+		"/path/to/helmfile.d": "/path/to/helmfile.d",
+	}
+	dirs := map[string]bool{
+		"helmfile.d": true,
+	}
+	files := map[string]string{
+		"helmfile.yaml": `
+environments:
+  prod:
+
+helmfiles:
+- helmfile.d/a*.yaml
+`,
+		"/path/to/helmfile.d/a1.yaml": `
+environments:
+  prod:
+
+releases:
+- name: zipkin
+  chart: stable/zipkin
+`,
+	}
+	globMatches := map[string][]string{
+		"/path/to/helmfile.d/a*.yaml": []string{"/path/to/helmfile.d/a1.yaml"},
+	}
+	fileExistsAt := func(path string) bool {
+		_, ok := files[path]
+		return ok
+	}
+	directoryExistsAt := func(path string) bool {
+		_, ok := dirs[path]
+		return ok
+	}
+	readFile := func(filename string) ([]byte, error) {
+		str, ok := files[filename]
+		if !ok {
+			return []byte(nil), fmt.Errorf("no file found: %s", filename)
+		}
+		return []byte(str), nil
+	}
+	glob := func(pattern string) ([]string, error) {
+		matches, ok := globMatches[pattern]
+		if !ok {
+			return []string(nil), fmt.Errorf("no file matched: %s", pattern)
+		}
+		return matches, nil
+	}
+	abs := func(path string) (string, error) {
+		a, ok := absPaths[path]
+		if !ok {
+			return "", fmt.Errorf("abs: unexpected path: %s", path)
+		}
+		return a, nil
+	}
+	app := &app{
+		readFile:          readFile,
+		glob:              glob,
+		abs:               abs,
+		fileExistsAt:      fileExistsAt,
+		directoryExistsAt: directoryExistsAt,
+		kubeContext:       "default",
+		logger:            helmexec.NewLogger(os.Stderr, "debug"),
+	}
+	noop := func(st *state.HelmState, helm helmexec.Interface) []error {
+		return []error{}
+	}
+
+	testcases := []struct {
+		name      string
+		expectErr bool
+	}{
+		{name: "undefined_env", expectErr: true},
+		{name: "default", expectErr: false},
+		{name: "prod", expectErr: false},
+	}
+
+	for _, testcase := range testcases {
+		err := app.FindAndIterateOverDesiredStates(
+			"helmfile.yaml", noop, "", []string{}, testcase.name,
+		)
+		if testcase.expectErr && err == nil {
+			t.Errorf("error expected but not happened for environment=%s", testcase.name)
+		} else if !testcase.expectErr && err != nil {
+			t.Errorf("unexpected error for environment=%s: %v", testcase.name, err)
+		}
+	}
+}

--- a/main.go
+++ b/main.go
@@ -616,7 +616,7 @@ func (a *app) FindAndIterateOverDesiredStates(fileOrDir string, converge func(*s
 			case *state.StateLoadError:
 				switch stateLoadErr.Cause.(type) {
 				case *state.UndefinedEnvError:
-					noMatchInThisHelmfile = true
+					continue
 				default:
 					return err
 				}

